src/Arrays/MPIStateArrays_kernels.jl

```diff
@@ -2,29 +2,28 @@ using Requires
 @init @require CUDAnative = "be33ccc6-a3ff-5ff2-a52e-74243cff1e17" begin
   using .CUDAnative
 end
+using KernelAbstractions.Extras: @unroll
 
-function knl_fillsendbuf!(::Val{Np}, ::Val{nvar}, sendbuf, buf,
-                          vmapsend, nvmapsend) where {Np, nvar}
+@kernel function knl_fillsendbuf!(::Val{Np}, ::Val{nvar}, sendbuf, buf,
+                                  vmapsend, nvmapsend) where {Np, nvar}
 
-  @inbounds @loop for i in (1:nvmapsend;
-                            threadIdx().x + blockDim().x * (blockIdx().x-1))
+  i = @index(Global, Linear)
+  @inbounds begin
     e, n = fldmod1(vmapsend[i], Np)
     @unroll for s = 1:nvar
       sendbuf[s, i] = buf[n, s, e]
     end
   end
-  nothing
```

Comment:
From the YAS Guide (https://github.com/jrevels/YASGuide)

> - The return keyword should always be omitted from return statements within short-form method definitions (f(...) = ...). The return keyword should never be omitted from return statements within any other context (function ... end, macro ... end, etc.).
> - If a function does not have a clearly appropriate return value, then explicitly return nothing.

If we are really planning on using the YAS guide for this project, can we please start following the advice on return statements?

Reply:
It is not easy to follow YAS here since the `@kernel` macro rewrites this function into three separate functions. After the macro has done its magic, `knl_fillsendbuf!` return a kernel object, and the kernel object returns an event. I would prefer if function annotated with `@kernel` do not have explicit return statements.

Reply:
Ah okay, that makes sense. I think leaving out the `return` statement in `@kernel` functions is fine as long as we document that somewhere and stick to it throughout the project.

Reply:
Also this made me think about what it means to have a return statement and I think KernelAbstraction will generally forbid that. https://github.com/JuliaGPU/KernelAbstractions.jl/issues/50